### DISCUSSION TITLE
Unable to parse IP address

### DISCUSSION THREAD

**Discussion by sebbbastien** [alecthomas/participle#234] [General]
Hi,

I'm unable to parse an IP address with a basic definition (Int.Int.Int.Int).

Here is a quick exemple: https://go.dev/play/p/s8XBKsabqXx

As far as I understand, it fails on the second dot, as if the float parser takes precedence.

Thanks in advance for your help!

**Comment by alecthomas** [comment]
Your intuition is correct, the default lexer will greedily match that input. You'll need a custom lexer.
